### PR TITLE
Use delete to disable automatic generated methods.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
Use delete to disable automatic generated methods instead of private, and put the constructor together for more clear.This modification cause the unused field warning, so add unused attribute to disable this warning.

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Had signed up the CLA,but seems not works.

**Comment by miasantreble** [comment]
Thanks for contributing to RocksDB! The changes look good to me, with a few recommendations:
* Please pick a more descriptive title for the PR since they will go to the commit message once merged.
* Currently there are 62 results from 47 files that contain the string "// No copying allowed", if you have time, do you mind applying the same changes to all these cases to make the code base more consistent?

**Comment by siying** [comment]
Can you improve the PR title?

**Comment by Shylock-Hg** [comment]
> Thanks for contributing to RocksDB! The changes look good to me, with a few recommendations:
> 
>     * Please pick a more descriptive title for the PR since they will go to the commit message once merged.
> 
>     * Currently there are 62 results from 47 files that contain the string "// No copying allowed", if you have time, do you mind applying the same changes to all these cases to make the code base more consistent?

Of course,I just want to know will you accept this change by this pull requests.I will correct all after.

**Comment by miasantreble** [comment]
All commit message in the PR will be squashed and replaced by the PR title and summary

File: db/db_iter.cc [hunk 1]
```diff
@@ -343,17 +347,19 @@ class DBIter final: public Iterator {
   LocalStatistics local_stats_;
   PinnedIteratorsManager pinned_iters_mgr_;
   ReadCallback* read_callback_;
+#ifdef ROCKSDB_LITE
```
Comment (miasantreble) [inline db/db_iter.cc]: why is this needed?

File: table/block_based_filter_block.h [hunk 2]
```diff
@@ -99,17 +102,13 @@ class BlockBasedFilterBlockReader : public FilterBlockReader {
 
  private:
   const FilterPolicy* policy_;
-  const SliceTransform* prefix_extractor_;
+  ROCKSDB_FIELD_UNUSED const SliceTransform* prefix_extractor_;
```
Comment (miasantreble) [inline table/block_based_filter_block.h]: what is this for?

File: db/db_iter.cc (hunk 1 above)
Comment (Shylock-Hg) [reply db/db_iter.cc]: Because this field is used just in LITE version.This will warn by compiler without UNUSED.

File: table/block_based_filter_block.h (hunk 2 above)
Comment (Shylock-Hg) [reply table/block_based_filter_block.h]: This will cause warning from compiler and will terminate compiling with -Werror.

**Comment by Shylock-Hg** [comment]
> All commit message in the PR will be squashed and replaced by the PR title and summary

Okay.I will provide the final title and summary.

**Comment by Shylock-Hg** [comment]
Final message:
Use `delete` to disable automatic generated methods.
Summary:
Use `delete` to disable automatic generated methods instead of `private`, and put the constructor together for more clear.This modification cause the *unused field* warning, so add unused attribute to disable this warning.

**Comment by miasantreble** [comment]
Hi @Shylock-Hg, apologize for the long delay, but do you mind rebasing from master and resolve the merge conflicts? 
I think the PR is almost ready to land except for the unused field which we can figure out once you resolve the conflicts.

**Comment by Shylock-Hg** [comment]
> Hi @Shylock-Hg, apologize for the long delay, but do you mind rebasing from master and resolve the merge conflicts?
> I think the PR is almost ready to land except for the unused field which we can figure out once you resolve the conflicts.

Maybe a new PR is simpler.

**Comment by siying** [comment]
@Shylock-Hg thank you so much for coming back to it. The Travis shows failure in LITE build, which needs to be fixed. The appveyor error is probably unrelated.

**Comment by Shylock-Hg** [comment]
> @Shylock-Hg thank you so much for coming back to it. The Travis shows failure in LITE build, which needs to be fixed. The appveyor error is probably unrelated.

Yes, I mark the unused field.

**Comment by siying** [comment]
The internal test failures don't seem to be related. They are simply timing out. I'm landing it and will monitor tests closely.
